### PR TITLE
Add annotations to buildkite builds

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
This brings the links to other builds up to a more accessible place
instead of only being buried in several dropdowns of logs:

presubmit-pipeline-ui-screenshot

### CODE REVIEW

File: build_tools/buildkite/scripts/wait_for_pipeline_success.py [hunk 1]
```diff
@@ -81,17 +87,28 @@ def main(args):
     build = bk.create_build()
   build_number = get_build_number(build)
   url = bk.get_url_for_build(build_number)
-  print(f"Waiting on {linkify(url)}")
+  ansi_link = ansi_linkify(url)
+  markdown_link = f"[{args.pipeline}]({url})"
+  print(f"Waiting on {ansi_link}")
+  if args.annotate:
+    # Don't explicitly set the context, so it can be set with an environment
```
Comment (pzread) [inline build_tools/buildkite/scripts/wait_for_pipeline_success.py:94]: Where will be the environment variable set?

File: build_tools/buildkite/pipelines/trusted/presubmit.yml [hunk 2]
```diff
@@ -38,4 +38,6 @@ steps:
     commands: |
       export BUILDKITE_ACCESS_TOKEN="$(gcloud secrets versions access latest \
           --secret=iree-buildkite-presubmit-pipelines)"
-      ./build_tools/buildkite/scripts/wait_for_pipeline_success.py build-runtime-cmake
+      ./build_tools/buildkite/scripts/wait_for_pipeline_success.py \
+        --annotate \
+        build-runtime-cmake
```
[the remark this answers is not in the document]
Comment (GMNGeoffrey) [reply build_tools/buildkite/pipelines/trusted/presubmit.yml:43]: Yes. Something like this (but with the text changed also):

image

File: build_tools/buildkite/scripts/wait_for_pipeline_success.py (hunk 1 above)
Comment (GMNGeoffrey) [reply build_tools/buildkite/scripts/wait_for_pipeline_success.py:94]: In the steps that call this, where I forgot to actually set it. Fixed, thank you :-)

File: build_tools/buildkite/pipelines/trusted/presubmit.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScottTodd) [reply build_tools/buildkite/pipelines/trusted/presubmit.yml:44]: Nice. That may get busy if we have 10 pipelines/steps, but we can iterate on it when that happens :)

**Comment by GMNGeoffrey** [comment]
> lgtm (after syncing past merge conflicts)

Amazing how it manages to have merge conflicts with the commit it was based on :facepalm:

**Review by pzread — APPROVED** [review]

File: build_tools/buildkite/pipelines/trusted/presubmit.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (GMNGeoffrey) [reply build_tools/buildkite/pipelines/trusted/presubmit.yml:44]: I think it's probably still better though. Like that's really the thing you look at and if you want more detail you can go down to the actual step runs. I really wanted some way to attach the link to each step, but I don't think there's a way to do that, unfortunately

**Pull request merged**